### PR TITLE
Implement mypy and fix static checking violations

### PR DESCRIPTION
**Pull request by alexrichey**
Running `mypy` over `dcpy` returns mostly type nit-picking, and also some real issues. We should consider implementing an automated mypy type check in CI, whenever files in `dcpy` are changed.

### CODE REVIEW

**Comment by fvankrieken** [comment]
Thanks for this, a lot of this is my doing. I would be very in favor of automatic checks, even if it's just adding to the overall requirements for now.

File: dcpy/utils/postgres.py [hunk 1]
```diff
@@ -5,20 +5,23 @@
...
 build_engine = create_engine(BUILD_ENGINE_RAW)
 
 
-def execute_query(query: str, placeholders: dict = None) -> None:
+def execute_query(query: str, placeholders: Optional[dict] = None) -> None:
     if placeholders is None:
         placeholders = {}
     with build_engine.connect() as connection:
-        connection.execute(statement=text(query), params=placeholders)
+        connection.execute(statement=text(query), parameters=placeholders)
```
Comment (alexrichey) [inline dcpy/utils/postgres.py:19]: FYI, this was one of the bugs. I haven't tested usages of this fn. We could benefit from some integration testing here!

**Comment by alexrichey** [comment]
> I would be very in favor of automatic checks, even if it's just adding to the overall requirements for now.

@fvankrieken (cc @damonmcc ) happy to just add these reqs to the standard requirements.[in/txt]. Thoughts?

**Comment by damonmcc** [comment]
yea the standard requirements seems good for now. love type checking!

**Review by fvankrieken — DISMISSED** [review]

File: .github/workflows/dcpy_test.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install mypy
+        run: pip install mypy
+
+      - name: (Temp) Install stubs
+        run: pip install -r python/requirements.txt
```
Comment (damonmcc) [inline .github/workflows/dcpy_test.yml:22]: why not add `mypy` to the requirements too?

File: pyproject.toml [hunk 3]
```diff
@@ -34,4 +34,8 @@ capitalisation_policy = "upper"
 [tool.pydocstyle]
 ignore = ["D100", "D101", "D102", "D107", "D104", "D213", "D407", "D413"]
 
-[tool.black]
\ No newline at end of file
+[tool.black]
+
+[[tool.mypy.overrides]]
+module = "geopandas.*" # no stubs available
+ignore_missing_imports = true
```
Comment (damonmcc) [inline pyproject.toml:41]: to confirm, we don't want `ignore_missing_imports` to be a global mypy setting?

I'm not familiar with configuring mypy but I think at the moment, this setting is module-specific to `geopandas` and it's submodules

Comment (alexrichey) [reply pyproject.toml:41]: @damonmcc Correct, we only want this to affect geopandas, which has no type stubs available (at least not in our current version). It's currently the only dcpy lib where that's the case.

File: .github/workflows/dcpy_test.yml (hunk 2 above)
Comment (alexrichey) [reply .github/workflows/dcpy_test.yml:22]: ah, good catch. Adding.

Comment (alexrichey) [reply .github/workflows/dcpy_test.yml]: @damonmcc Added

**Review by damonmcc — APPROVED** [review]

**Review by fvankrieken — APPROVED** [review]

**Pull request merged**